### PR TITLE
Improved gather points for lineplots

### PR DESCRIPTION
**Pull request by corranwebster**
This adds optimized numpy routines for detecting when intervals may intersect the visible range.  This significantly speeds up calculations for large datasets.

### CODE REVIEW

**Comment by corranwebster** [comment]
This is also ready for review.

File: chaco/base.py [hunk 1]
```diff
@@ -226,4 +245,57 @@ def point_line_distance(pt, p1, p2):
...
+
+    This is an optimized routine for detecting which points are endpoints
+    of visible segments in a 1D polyline.  An optional mask can be provided for
+    points which should be excluded from consideration for other reasons
+    (such as not being selected).  Returns a mask of points which are
+    endpoints of intervals which potentially intersect the range.
+
+    Parameters
+    ----------
+    x : 1d array
```
Comment (jvkersch) [inline chaco/base.py]: I'm having some trouble understanding how the entries of `x` are related to the intervals. If `x = [A, B, C, ...]`, are the intervals `[A, B], [B, C], ...`?

Comment (corranwebster) [inline chaco/base.py]: Yes.

Comment (jvkersch) [inline chaco/base.py]: Can you add a note of this to the docstring?

**Comment by corranwebster** [comment]
Merging

**Pull request merged**